### DISCUSSION TITLE
Adafuit / UDA1334A Rout and Lout

### DISCUSSION THREAD

**Discussion by jmsunseri** [pschatzmann/ESP32-A2DP#458] [Q&A] [answered]
Would it be possible to connect the Rout and Lout from a UDA1334A to the esp32 and use that as a source for sound data?

**Comment by pschatzmann** [comment]
You would need a proper audio adc. I suggest you have a look at the external adc and audio board wikis of https://github.com/pschatzmann/arduino-audio-tools/wiki

**Reply by jmsunseri** [reply]
My goal is to take the audio data from a Nintendo Gameboy Advance SP which had L-out and R-out points to connect to and send to a bluetooth sink.  I've seen other projects which used a KCX_BT_EMITTER module to do this but i wanted something a little smarter and more configurable.  The touch sensor on the ESP32 is a big bonus because i should be able to control it without cutting a hole in the shell.  I really don't know anything about audio.  I just briefly looked at the external ADC and audio boards wiki pages and I didn't see anything immediately applicable taking a L-out and R-out connection and uses them as a music data source.

**Comment by pschatzmann** [comment]
I suggest to look again. All devices/modules that have a 3.5mm or RCA audio input should potentially work.

**Reply by jmsunseri** [reply]
are you suggesting that i need an extra module to make this work or that those modules make for a good example on how to handle this?

**Comment by pschatzmann** [comment]
Yes, you need the proper hardware to input audio in stereo with a good quality

**Reply by jmsunseri** [reply]
The audio coming out of a gameboy is not what i would consider to be high quality.  I might give this a try https://docs.espressif.com/projects/esp-idf/en/v4.2/esp32/api-reference/peripherals/adc.html#application-examples I'm assuming that the `adc1_get_raw` method might return me an object that matches the format of `StarWars10_raw` in your example.

**Comment by pschatzmann** [comment, marked as answer]
I think, no: You would need to use the AnalogAudioStream (of the AudioTools) which uses the esp32 ADC via the I2S DMA API to get the necessary sample rate. However it supports only mono. Or you can use the IDF I2S API directlyy.

But expect quite some noise...

**Reply by jmsunseri** [reply]
The gameboy only had one speaker.  maybe mono would be just fine.  maybe adding a resistor to the pad would be sufficient as described here?  https://docs.espressif.com/projects/esp-idf/en/v4.2/esp32/api-reference/peripherals/adc.html#minimizing-noise

**Reply by pschatzmann** [reply]
Please keep me updated on your findings.

Just one more word of warning: While the A2DP Sink is easy to implement and stable, the A2DP Source scenario has quite some challenges and is not so easy.

**Reply by jmsunseri** [reply]
Will do that. Maybe i can make more progress tomorrow after work.

**Comment by pschatzmann** [comment]
https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-basic-api/base-adc-a2dp might be a good starting point

**Reply by jmsunseri** [reply]
not only did this work, it worked exceptionally well!!!  pulled the sound right off positive heading into the single speaker and it worked like a charm with absolutely no static.  didn't even need the capacitor.  my only issue so far is that the volume control affected the volume heading to both the bluetooth device and the actual speaker.  when they turn on the bluetooth they are probably going to turn their volume down on the speaker so i have some thinking to do but i think this is going to work out really well.
